### PR TITLE
[DNM] cluster: add advertise addr for tidb,tikv,pd

### PR DESCRIPTION
**Pull request by 9547**
### What problem does this PR solve? 

implement https://github.com/pingcap/tiup/issues/1218

- [ ] Need more test case
- [ ] Add example config inside templates

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
NONE
```

### CODE REVIEW

**Comment by lucklove** [comment]
Hold on please, this is a critical change on product level... we are talking about this now

**Review by AstroProfundis — COMMENTED** [review]
We should also add default values to these fields in `spec`, as most users don't need to customize them at all. And maybe some helper functions to get the address from an instance object.

File: pkg/cluster/template/scripts/pd.go [hunk 1]
```diff
@@ -77,6 +84,36 @@ func (c *PDScript) WithPeerPort(port int) *PDScript {
 	return c
 }
 
+// WithAdvertisWithAdvertiseClientAddr set AdvertiseClientAddr field of PDScript
```
Comment (lucklove) [inline pkg/cluster/template/scripts/pd.go:87]: The comment is not correct

**Comment by lucklove** [comment]
Support in https://github.com/pingcap/tiup/pull/1224

**Pull request closed without merging**